adam-core/src/main/scala/org/bdgenomics/adam/rdd/ShuffleRegionJoin.scala

```diff
@@ -38,488 +35,387 @@ import scala.reflect.ClassTag
 sealed abstract class ShuffleRegionJoin[T: ClassTag, U: ClassTag, RT, RU]
     extends RegionJoin[T, U, RT, RU] {
 
-  protected val sd: SequenceDictionary
-  protected val partitionSize: Long
-  protected val sc: SparkContext
+  // stores join candidate tuples from right
+  protected val cache: ListBuffer[(ReferenceRegion, U)] = ListBuffer.empty
```

Comment:
I pushed all of this up to the parent class to get rid of 2 levels of abstraction we were previously using, as well as some boiler-plate copy/pasted code that didn't need to be there. This also reduces the size of objects and overhead cost of `shuffleRegionJoins`. Performance numbers coming.